nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -105,4 +112,20 @@ Controller::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameters)
   return result;
 }
 
+geometry_msgs::msg::Twist Controller::rotateToTarget(const double & angle_to_target)
+{
+  geometry_msgs::msg::Twist vel;
+  vel.linear.x = 0.0;
+  vel.angular.z = 0.0;
+  if(angle_to_target > 0) {
+    vel.angular.z = std::clamp(1.0 * angle_to_target * v_angular_max_,
```

Comment:
`1.0 *` in both expressions seems unnecessary. I think the other one might mean `-1.0`. 

What is the derivation of this rotation code for `angle_to_target * v_angular_max`? Might it be better to steal the rotation code from something like the rotation shim controller / spin behavior to make this consistent?

Reply:
Odometric error is often large with high acceleration, so it would probably be wise to have acceleration / decelleration in the rotation action such that we minimize slippage -- since especially in this case, we're fully dead reckoning

Reply:
@SteveMacenski How should I take the current velocity inside docking_server in the good way?
shim controller requires it 
https://github.com/ros-navigation/navigation2/blob/f426bdaadaa1a124bea0da5f32961cef6d679ae2/nav2_rotation_shim_controller/src/nav2_rotation_shim_controller.cpp#L288

Reply:
You could get it the same way we do in the controller server with an odometry subscriber: https://github.com/ros-navigation/navigation2/blob/f426bdaadaa1a124bea0da5f32961cef6d679ae2/nav2_controller/src/controller_server.cpp#L626

Reply:
Thank you @SteveMacenski, it's already implemented and I hope I managed it well.

Reply:
I just noticed that from time to time (not in 100% of situations) robot stops during performing an initial rotation. The angular distance is correct but robot the generatated velocities are like 0.003 rad/s
Now I am inspecting why that happens.